PSReadLine/Keys.cs

```diff
@@ -149,11 +149,20 @@ internal static void TryGetCharFromConsoleKey(ConsoleKeyInfo key, ref char resul
             }
             int charCount = ToUnicode(virtualKey, scanCode, state, chars, chars.Length, flags);
 
-            // TODO: support diacriticals (charCount == 2)
             if (charCount == 1)
             {
                 result = chars[0];
             }
+            else if (charCount == -1 || charCount >=2)
```

Comment:
You'll want to test starting PowerShell (fresh new window so new conhost) with different layouts, e.g.

1. layout in en-us
2. start powershell
3. switch layout

The problem is some apis like `ToUnicode` (I think) don't honor the current keyboard layout - only the layout that was active when starting conhost.

Reply:
Just tried this on my french locale machine (came with fr-FR keyboard by default, and installed English (US) keyboard layout). It works fine.

Reply:
Here is the screen record. It looks `ToUnicode` does honor the current keyboard layout.

![a](https://user-images.githubusercontent.com/127450/59050967-9a542100-8840-11e9-92bf-b71c7ebee47d.gif)

Reply:
The tests passed on the French machine:

![image](https://user-images.githubusercontent.com/127450/59051564-08e5ae80-8842-11e9-8eb5-a8c614305f96.png)